app/Http/Controllers/API/CodeSnippetController.php

```diff
@@ -18,29 +20,99 @@ class CodeSnippetController extends Controller
      */
     public function getGistWidget(Request $request, DataGrabber $grabber)
     {
-        $link = $request->get('link') . '.json';
-        $prefix = env('SERVER_PREFIX', '');
-        $jsPath = url(($prefix ? '/' : '' ) . env('JS_PATH'));
+        // Validating the input
+        $validator = Validator::make(['link' => $request->get('link')], [
+            'link' => [
+                'required',
+                'regex:/(ftp|http|https):\/\/gist.github.com\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-\/]))?/i'
+            ]
+        ]);
 
-        if ($prefix) {
-            $prefixToPaste = $prefix . '/';
-        } else {
-            $prefixToPaste = '';
+        if ($validator->fails()) {
+            return $this->returnWidgetError('Incorrect Gist path');
         }
 
+        // Trying to get a data from remote server
+        $link = $request->get('link') . '.json';
+
         try {
             $data = $grabber->getFromJson($link);
         } catch (RemoteDataGrabberException $e) {
-            return Response::view('errors.gist.404', [
-                'js_path'  => $jsPath,
-                'prefix'  => $prefixToPaste,
-            ],  404);
+            return $this->returnWidgetError('GitHub Gist is not found');
         }
 
+        // Getting a prefix and jsPath
+        $prefix = env('SERVER_PREFIX', '');
+        $jsPath = url(($prefix ? '/' : '' ) . env('JS_PATH'));
+
+        // Successfull answer with gist
         return Response::view('widgets.gist', [
             'stylesheet_link' => $data->stylesheet,
             'snippet'         => $data->div,
-            'js_path' => $jsPath,
+            'js_path'         => $jsPath,
         ], 200);
     }
-}
+
+    public function getPastebinWidget(Request $request, DataGrabber $grabber)
+    {
+        $link = $request->get('link');
+
+        // Validating the input
+        $validator = Validator::make(['link' => $link], [
+            'link' => [
+                'required',
+                'regex:/(ftp|http|https):\/\/pastebin.com\/embed_iframe.php\?i=[a-z0-9]+/i'
+            ]
+        ]);
+
+        if ($validator->fails()) {
+            return $this->returnWidgetError('Incorrect Pastebin path');
+        }
+
+        try {
+            $data = $grabber->getRaw($link);
+        } catch (RemoteDataGrabberException $e) {
+            return $this->returnWidgetError('Pastebin snippet is not found');
+        }
+
+        // Getting a prefix and jsPath
+        $prefix = env('SERVER_PREFIX', '');
+        $jsPath = url(($prefix ? '/' : '' ) . env('JS_PATH'));
```

Comment:
I thought about it, but if we isolate a validation, there the same amount of code, so no sense. As for jsPath, I added one.